test/e2e/testing-manifests/auth/encrypt/kind.yaml

```diff
@@ -21,8 +21,39 @@ nodes:
     readOnly: true
     propagation: None
   kubeadmConfigPatches:
+    # v1beta4 for the future (v1.35.0+ ?)
+    # https://github.com/kubernetes-sigs/kind/issues/3847
+    # TODO: drop v1beta3 when kind makes the switch
     - |
       kind: ClusterConfiguration
+      apiVersion: kubeadm.k8s.io/v1beta4
+      apiServer:
+        extraArgs:
+          - name: "encryption-provider-config"
+            value: "/etc/kubernetes/encryption-config.yaml"
+          - name: "v"
+            value: "5"
+        extraVolumes:
+        - name: encryption-config
+          hostPath: "/etc/kubernetes/encryption-config.yaml"
+          mountPath: "/etc/kubernetes/encryption-config.yaml"
+          readOnly: true
+          pathType: File
+        - name: sock-path
+          hostPath: "/tmp"
+          mountPath: "/tmp"
+      scheduler:
+        extraArgs:
+          - name: "v"
+            value: "5"
+      controllerManager:
+        extraArgs:
+          - name: "v"
+            value: "5"
+    # v1beta3 for v1.23.0 ... ?
```

Comment:
1.33?  (v1beta3 )

```
//   - kubeadm v1.31.x and newer can be used to migrate from v1beta3 to v1beta4.
```

https://github.com/kubernetes/kubernetes/blob/6822f9b0af2ae1633739a5ba561938f72f470cdf/cmd/kubeadm/app/apis/kubeadm/v1beta4/doc.go#L62-L67

Reply:
No, these are the Kubernetes version for which kind uses the kubeadm config API version.

1.23.0 to current use v1beta3 right now

presumably these configs don't need to support <1.23.0, but they might've worked previously

for 1.x?+ we will use v1beta4 (see the linked discussion)

Reply:
Yeah. We were much quicker last time, but also we were able to stay on version-independent patches for v1beta, v1beta2, v1beta3, v1beta4 is the first time we really need distinct configurations. Also at the moment we have mostly been fielding bug fixes, kind hasn't had a lot of active feature development, so this slipped.

I'm hoping we can sort this out quickly and not be an obstacle.